### PR TITLE
Initial layout saving implemented. Code is gross.

### PR DESCRIPTION
**Pull request by jonahdlin**
We got basic layout saving and recalling set up. Works without any bugs. The frontend code is pretty hideous and it's not in any of the right places. I'll work on setting it up properly when I do more refactoring.

### CODE REVIEW

File: src/client/features/view/index.js [hunk 1]
```diff
@@ -89,8 +99,33 @@ class View extends React.Component {
...
+    }
+
     cy.nodes('[class="complex"], [class="complex multimer"]').filter(node => node.isExpanded()).collapse();
-    cy.layout(lo.layoutMap.get(layoutName)).run();
+    let layout = cy.layout(lo.layoutMap.get(layoutName));
+    let that = this;
+    layout.pon('layoutstop').then(function() {
+      if (that.props.admin && layoutName !== lo.humanLayoutName) {
+        let posObj = {};
+        for (let i = 0; i < 25; i++) {
```
Comment (d2fong) [inline src/client/features/view/index.js]: What is the purpose of iterating 25 times here?

Comment (jonahdlin) [reply src/client/features/view/index.js]: Accidentally left in a hard coded value. Just fixed it.

File: src/client/features/view/components/graph/layout/index.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ const stratifiedLayeredOpts = require('./stratifiedKlay');
 
 const coseBilkentMaxGraphSize = 100;
 
+const humanLayoutName = 'Human-created';
```
Comment (d2fong) [inline src/client/features/view/components/graph/layout/index.js]: Not sure it makes sense for this to be here.

File: src/client/features/view/cy/events/move.js [hunk 3]
```diff
@@ -1,10 +1,9 @@
-const CDC = require('../../../../services/').CDC;
+const CDC = require('../../../../services').CDC;
 
 const bindMove = (uri, version, cy) => {
```
Comment (d2fong) [inline src/client/features/view/cy/events/move.js]: TODO: Ask Max if there is a better pattern for binding all of these events.

File: src/client/features/view/index.js [hunk 4]
```diff
@@ -89,8 +99,34 @@ class View extends React.Component {
...
+    }
+
     cy.nodes('[class="complex"], [class="complex multimer"]').filter(node => node.isExpanded()).collapse();
-    cy.layout(lo.layoutMap.get(layoutName)).run();
+    let layout = cy.layout(lo.layoutMap.get(layoutName));
+    let that = this;
+    layout.pon('layoutstop').then(function() {
+      if (that.props.admin && layoutName !== lo.humanLayoutName) {
+        let posObj = {};
+        let numNodes = cy.nodes().length;
```
Comment (d2fong) [inline src/client/features/view/index.js]: This can be simplified to:
```js
cy.nodes().forEach(node => {
   posObj[node.id()] = node.position();
})

```

**Pull request merged**